lib/groups_page.rb

```diff
@@ -4,9 +4,21 @@
 class GroupsPage < SobraniePage
   decorator Scraped::Response::Decorator::AbsoluteUrls
 
+  CEASED_MEMBERS_URL = 'mps-whose-mandate-has-not-been-completed-2014-2018'
+
   field :groups do
-    noko.css('div.toc li a').map do |a|
-      fragment a => LinkLine
-    end
+    # The rejected page is not a list of members grouped by party
+    groups_links.reject { |a| a.attr('href').include? CEASED_MEMBERS_URL }
+                .map { |a| fragment a => LinkLine }
+  end
+
+  field :ceased_members_url do
+    groups_links.find { |a| a.attr('href').include? CEASED_MEMBERS_URL }.attr('href')
+  end
```

Comment:
There's a little too much repetition here. I'd suggest instead that you `partition` on that, along the lines of:

```ruby
  field :groups do
    groups_links.last.map { |a| fragment a => LinkLine }
  end

  field :ceased_members_url do
    groups_links.first.first.attr('href')
  end

  private

  # The layout of this page is to list all the groups, but at the end,
  # in an otherwise undistinguished entry, there's a list to a separate
  # page of all the ceased members. 
  def groups_links
    noko.css('div.toc li a').partition { |a| a.attr('href').include? CEASED_MEMBERS_URL }
  end
```